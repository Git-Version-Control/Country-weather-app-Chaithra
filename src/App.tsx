import { Routes, Route, Link } from "react-router-dom";
import CountryDetails from "./pages/CountryDetails";
import NotFound from "./pages/NotFound";
import styles from "./App.module.css";
import InputPage from "./pages/InputPage";

const myStyle={
  backgroundImage: "url(./types/bkgd.png)",
  height:'100vh',
  marginTop:'-70px',
  fontSize:'50px',
  backgroundSize: 'cover',
  backgroundRepeat: 'no-repeat',
};

function App() {
  return (
    <div  style={myStyle}>
      <Link className={styles.header} to="/">
        Country Weather App
      </Link>
      <div className={styles.appContainer}>
        <Routes>
          <Route path="/" element={<InputPage/>} />
          <Route path="countryDetails" element={<CountryDetails />} />
          <Route path="*" element={<NotFound />} />
        </Routes>
      </div>
    </div>
  );
}

export default App;
