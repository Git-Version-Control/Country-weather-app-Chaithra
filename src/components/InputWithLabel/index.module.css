.container {
    display: flex;
    flex-flow: column wrap;
  }
  
  /*.label {
    font-size: x-large;
    margin-bottom: 1rem;
    color:rgb(12, 12, 154);
    font-style: bold;
    font-weight: 900;
  }*/
  
  .input {
    padding: 2rem;
    font-size: 3rem;
    border-radius: 8px;
    height:0.8em;
  }
  