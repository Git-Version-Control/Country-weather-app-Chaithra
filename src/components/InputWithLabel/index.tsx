
import { style } from "@mui/system";
import { ChangeEvent, FC } from "react";
import styles from "./index.module.css";

type InputWithLabelProps = {
  //labelName: string;
  id: string;
  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
  value: string;
  placeholder: string;
};

const InputWithLabel: FC<InputWithLabelProps> = ({
 // labelName,
  id,
  value,
  placeholder,
  onChange,
}) => (
  <div className={styles.container}>
      
    <input
      className={styles.input}
      id={id}
      onChange={onChange}
      value={value}
      placeholder={placeholder}
    />
  </div>
);

export default InputWithLabel;