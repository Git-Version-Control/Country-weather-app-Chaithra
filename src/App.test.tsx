
import { render, screen, fireEvent, act } from "@testing-library/react";

import App from "./App";
import { BrowserRouter } from "react-router-dom";
import InputWithLabel from "./components/InputWithLabel";
import axios from "axios";
import { title } from "process";

describe("unit testing basics", () => {
  test("assert if title is React", () => {
    expect(true).toBe(true);
  });

});