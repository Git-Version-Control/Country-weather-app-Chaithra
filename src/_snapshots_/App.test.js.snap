// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Input with label component snapshot testing 1`] = `
<div>
  <input
    type="text"
    value="India"
  />
  
</div>
`